addons/web/static/src/core/barcode/barcode_video_scanner.js

```diff
@@ -77,6 +78,12 @@ export class BarcodeVideoScanner extends Component {
                 this.props.onError(new Error(errorMessage));
                 return;
             }
+            if (!this.videoPreviewRef.el) {
```

Comment:
Is there something special with `<video>` elements? From an owl point of view, the ref must exist here because the element isn't conditionally displayed

Reply:
If the component is unmounted while this part of code is still going on (for example, the user clicks on the button to display this component and clicks directly on the button again to stop the video and remove the component), `this.videoPreviewRef.el` is `null` which causes issue with the following code.

Reply:
damn you're right, I missed the async part